gprofiler/main.py

```diff
@@ -230,8 +239,23 @@ def _strip_container_data(collapsed_data):
             lines.append(line[line.find(';') + 1 :])
         return '\n'.join(lines)
 
+    def send_metadata(self):
+        cloud_metadata = get_cloud_instance_metadata()
+        system_metadata = get_system_info()
+        metadata_dict = {
+            "cloud_provider": cloud_metadata.pop("provider") or "unknown",
+            "version": __version__,
+            "start_time": round(self._spawn_time),
+        }
+        metadata_dict.update(system_metadata.get_dict())
+        if cloud_metadata is not None:
+            metadata_dict["cloud_info_wrapped"] = cloud_metadata
+        self._client.submit_metadata(metadata_dict)
+
     def start(self):
         self._stop_event.clear()
+        if self._system_metrics_monitor is not None:
```

Comment:
To avoid all the `is None` checks, I think a nicer impl is to create a dummy class (like `NoopProfiler`) that returns `None`  when you call your methods